clang/test/CodeGenSYCL/stall_enable_device.cpp

```diff
@@ -2,20 +2,21 @@
 
 // Tests for IR of Intel FPGA [[intel::use_stall_enable_clusters]] function attribute on Device.
 // The metadata to be attached to the functionDecl that the attribute is applied to.
-// The attributes do not get propagated to kernel metadata i.e. spir_kernel.
+// The attributes get propagated to the kernel metadata i.e. spir_kernel if directly applied
```

Comment:
I don't think this comment is required. This is standard behavior for SYCL attributes, which is why I was surprised the behavior was different for this attribute in https://github.com/intel/llvm/pull/3900. You can probably just say - `// Tests for IR of Intel FPGA [[intel::use_stall_enable_clusters]] function attribute on Device`